src/Slackbot.Net.Extensions.FplBot/Handlers/FplSearchHandler.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using Fpl.Search.Searching;
+using Slackbot.Net.Endpoints.Abstractions;
+using Slackbot.Net.Endpoints.Models.Events;
+using Slackbot.Net.Extensions.FplBot.Abstractions;
+using Slackbot.Net.Extensions.FplBot.Helpers;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Fpl.Client.Abstractions;
+using Microsoft.Extensions.Logging;
+using Slackbot.Net.Extensions.FplBot.Extensions;
+
+namespace Slackbot.Net.Extensions.FplBot.Handlers
+{
+    public class FplSearchHandler : HandleAppMentionBase
+    {
+        private readonly ISearchClient _searchClient;
+        private readonly IGameweekClient _gameweekClient;
+        private readonly ISlackWorkSpacePublisher _workSpacePublisher;
+        private readonly ILogger<FplSearchHandler> _logger;
+
+
+        public FplSearchHandler(
+            ISearchClient searchClient,
+            IGameweekClient gameweekClient,
+            ISlackWorkSpacePublisher workSpacePublisher, 
+            ILogger<FplSearchHandler> logger)
+        {
+            _searchClient = searchClient;
+            _gameweekClient = gameweekClient;
+            _workSpacePublisher = workSpacePublisher;
+            _logger = logger;
+        }
+
+        public override string[] Commands => new[] { "search" };
+
+        public override async Task<EventHandledResponse> Handle(EventMetaData eventMetadata, AppMentionEvent message)
+        {
+            var term = ParseArguments(message);
+
+            var entriesTask = _searchClient.SearchForEntry(term, 10);
+            var leaguesTask = _searchClient.SearchForLeague(term, 10);
+
+            var entries = await entriesTask;
+            var leagues = await leaguesTask;
```

Comment:
Why not inline the awaits..?

Reply:
Really? Ok, I thought there was no difference to this approach vs just the inlined await version. I would have thought line 45 kicked it off. 🤷‍♂️

Reply:
Yes, really 😄 

```csharp
static async Task Main(string[] args)
{
    await Task1();
    await Task2();
    await Task3();
}
```
👇 

![1](https://user-images.githubusercontent.com/661484/104930806-ecaa6d00-59a5-11eb-92c2-b6ee2590e9c6.gif)

```csharp
static async Task Main(string[] args)
{
    var task1 = Task1();
    var task2 = Task2();
    var task3 = Task3();

    await task1;
    await task2;
    await task3;
}
```
👇 

![2](https://user-images.githubusercontent.com/661484/104930837-f207b780-59a5-11eb-85a0-cb18e1ca9e92.gif)

```csharp
static async Task Task1()
{
    Console.WriteLine("Starting Task 1");
    await Task.Delay(1000);
    Console.WriteLine("Task 1 done");
}

static async Task Task2()
{
    Console.WriteLine("Starting Task 2");
    await Task.Delay(2000);
    Console.WriteLine("Task 2 done");
}

static async Task Task3()
{
    Console.WriteLine("Starting Task 1");
    await Task.Delay(3000);
    Console.WriteLine("Task 3 done");
}
```

Reply:
Ah, you mean paralel-ish, then 👍 I guess you'd want Task.WhenAll, unless you always know Task1 is quicker than the rest.

Reply:
Yes, I know. This will do the same thing, but it's not more readable imo:

```csharp
static async Task Main(string[] args)
{
    var result = await Task.WhenAll(Task1(), Task2(), Task3());
    
    var first = result[0];
    var second = result[1];
    var third = result[2];
}
```
Or in our case 👇 
```csharp
var result = await Task.WhenAll(_searchClient.SearchForEntry(term, 10), _searchClient.SearchForLeague(term, 10));

var entries = result[0];
var leagues = result[1];
```

But I guess it's clearer that they will happen in parallel here ☝️ (or parallel-ish, which is what Task.WhenAll does as well - it fires them away in the order they are given). But more ugly tho. No? 🤷

Reply:
Yeah, not much to gain, agree